src/EnergyPlus/UnitarySystem.cc

```diff
@@ -18173,12 +18173,11 @@ namespace UnitarySystems {
 
     void isWaterCoilHeatRecoveryType(EnergyPlusData const &state, int const waterCoilNodeNum, bool &nodeNotFound)
     {
-        for (auto &unitarySystem : state.dataUnitarySystems->unitarySys) {
-            if (unitarySystem.m_HRcoolCoilFluidInletNode == waterCoilNodeNum && unitarySystem.m_WaterHRPlantLoopModel) {
-                nodeNotFound = false;
-                break;
-            }
-        }
+        if (!nodeNotFound) return;
+        nodeNotFound = std::none_of(
```

Comment:
Nice!   OK, so return early if nodeNotFound is _already_ set to false -- indicating the node has already been found.  If the node may still be not found, then just set it to the value of a none-of lookup.

Reply:
It is true that if nodeNotFound is false it has already been found and will not be found in the following search. I could not figure out a way to get this to work without this return. If nodeNotFound is passed in as false, then the next search of std::none_of will return true because none of the unitary systems has that combination (i.e., waterCoilNodeNum was found elsewhere in a coil not wrapped in a unitary sytem).

Reply:
I don't not misunderstand this.